src/libraries/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -89,9 +80,16 @@ public string Method
             get { return _method; }
         }
 
-        internal byte[]? Http3EncodedBytes
+        internal byte[] Http3EncodedBytes
         {
-            get { return _http3EncodedBytes; }
+            get {
+                if (_http3EncodedBytes != null)
+                    return _http3EncodedBytes;
+
+                return _http3EncodedBytes = _http3Index is int index && index >= 0 ?
+                    QPackEncoder.EncodeStaticIndexedHeaderFieldToArray(index) :
+                    QPackEncoder.EncodeLiteralHeaderFieldWithStaticNameReferenceToArray(H3StaticTable.MethodGet, _method);
```

Comment:
This similarly doesn't look thread-safe but needs to be (this instance may be a singleton shared amongst multiple threads).  The runtime will likely include a store barrier on the write to _http3EncodedBytes, but it'd be better to be explicit about it:
```C#
internal byte[] Http3EncodedBytes
{
    get
    {
        byte[]? http3EncodedBytes = Volatile.Read(ref _http3EncodedBytes);
        if (http3EncodedBytes is null)
        {
            http3EncodedBytes = _http3Index is int index && index >= 0 ?
                PackEncoder.EncodeStaticIndexedHeaderFieldToArray(index) :
                QPackEncoder.EncodeLiteralHeaderFieldWithStaticNameReferenceToArray(H3StaticTable.MethodGet, _method);
            Volatile.Write(ref _http3EncodedBytes, http3EncodedBytes);
        }

        return http3EncodedBytes;
    }
}

Reply:
(Also, the old implementation would be inlined due to being trivial.  The new will not.  I don't know if this property is on any hot paths.)

Reply:
Why does it need to be thread-safe? Its an internal property which if there is a race would return the same data but in a different instance.

Reply:
In case the writes to populate the instance were reordered after the write to publish the instance.

Reply:
Addressed